common/index.ts

```diff
@@ -44,5 +44,6 @@ export enum AuthType {
  */
 export function isValidResourceName(resourceName: string): boolean {
   // see: https://javascript.info/regexp-unicode
-  return !/[\p{C}%]/u.test(resourceName) && resourceName.length > 0;
+  const exp = new RegExp('[p{C}%]', 'u');
```

Comment:
I think this is going to match the `p` character, I think you need `\p{C}` if you want to make sure that we are looking for unicode characters